mongorestore/mongorestore_test.go

```diff
@@ -583,3 +583,91 @@ func TestFixHashedIndexes(t *testing.T) {
 		})
 	})
 }
+
+func TestRandom(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	ctx := context.TODO()
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	Convey("Test MongoRestore with {autoIndexId: false} in a non-local database's collection", t, func() {
+		dbName := session.Database("testdata")
+
+		defer dbName.Collection("test_auto_idx").Drop(ctx) // drop the collection to clean up resources
```

Comment:
I don't remember how exactly it all works under the hood, but I think goconvey does some funny things with the closures you pass to it. I'm not sure whether a defer is safe or not in a convey block.

IIRC, there's some sort of special way to declare cleanup code that guarantees it will play nicely with the rest of convey.

Reply:
Discussed offline: defers seem to work inside of conveys used by other mongorestore tests, so it should be fine.